### PR TITLE
[KommunetvIE] Add extractor for kommunetv.no

### LINKED ISSUE #30801: Add support for kommunetv.no
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://oslo.kommunetv.no/archive/921
- Single video: https://rogfk.kommunetv.no/archive/54
- Playlist: not relevant

## Description

WRITE DESCRIPTION HERE
This website is used by Norwegian counties for distribution of political meetings. The content is not under copyright. Does not require login credentials.
The website compiles many small videos and multiple camera angles into a presentation playlist. In many ways like mediasite.com. But i dont know if its from a technical perspective the same thing.
Each county uses its own subdomain, but the url structure is the same.
At this time youtube-dl errors with "Unsupported URL"

### PR DESCRIPTION
**Pull request by fonkap**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
Adds support for kommunetv.no

Fixes #30801

### CODE REVIEW

**Comment by fonkap** [comment]
This extractor only downloads the first camera. I don't know a proper way to handle multiple cams.

**Review by dirkf — CHANGES_REQUESTED** [review]
Thanks for your work!

Just a few nits. Then check the CI tests and should be OK to merge.

File: youtube_dl/extractor/kommunetv.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        headers = {
+            'Accept': 'application/json'
+        }
+        data = self._download_json('https://oslo.kommunetv.no/api/streams?streamType=1&id=%s' % video_id, video_id, headers=headers)
+        title = data['stream']['title']
+        file = data['playlist'][0]['playlist'][0]['file']
+        url = self._search_regex(r'(^[^\?]*)', file, 'url')
```
[the remark this answers is not in the document]
Comment (fonkap) [reply youtube_dl/extractor/kommunetv.py:27]: nice!

[the remark this answers is not in the document]
Comment (fonkap) [reply youtube_dl/extractor/kommunetv.py]: oops there is a problem with this.
base_url removes the filename

**Comment by fonkap** [comment]
sorry, wrong commit msg, should be:
[KommunetvIE] revert base_url to regex

**Review by fonkap — COMMENTED** [review]
base_url drops the filename before the ? in the url

File: youtube_dl/extractor/kommunetv.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirkf) [reply youtube_dl/extractor/kommunetv.py:27]: What does the `file` look like?

**Comment by dirkf** [comment]
> sorry, wrong commit msg, should be: [KommunetvIE] revert base_url to regex

We'll squash the commits in the merge anyway, but feel free to rebase and push again.

File: youtube_dl/extractor/kommunetv.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirkf) [reply youtube_dl/extractor/kommunetv.py:27]: Oh, yes, I should know (#31490), sorry.

As of today try this instead (you may need to sync from the latest master):
`url = update_url(file, query=None, fragment=None)`

**Review by fonkap — COMMENTED** [review]
Using update_url.
Works fine, thank you very much!

**Review by dirkf — APPROVED** [review]

**Pull request merged**